habitat-lab/habitat/datasets/rearrange/rearrange_generator.py

```diff
@@ -391,29 +393,15 @@ def generate_scene(self) -> str:
 
     def visualize_scene_receptacles(self) -> None:
         """
-        Generate a wireframe bounding box for each receptacle in the scene, aim the camera at it and record 1 observation.
+        Generate a debug line representation for each receptacle in the scene, aim the camera at it and record 1 observation.
         """
         logger.info("visualize_scene_receptacles processing")
         receptacles = find_receptacles(self.sim)
         for receptacle in receptacles:
             logger.info("receptacle processing")
-            viz_objects = receptacle.add_receptacle_visualization(self.sim)
-
-            # sample points in the receptacles to display
-            # for sample in range(25):
-            #     sample_point = receptacle.sample_uniform_global(self.sim, 1.0)
-            #     sutils.add_viz_sphere(self.sim, 0.025, sample_point)
-
-            if viz_objects:
-                # point the camera at the 1st viz_object for the Receptacle
-                self.vdb.look_at(
-                    viz_objects[0].root_scene_node.absolute_translation
-                )
-                self.vdb.get_observation()
-            else:
-                logger.warning(
-                    f"visualize_scene_receptacles: no visualization object generated for Receptacle '{receptacle.name}'."
-                )
+            receptacle.debug_draw(self.sim)
+            self.vdb.look_at(receptacle.sample_uniform_global(self.sim, 1.0))
```

Comment:
Receptacle does not have a "position" per-se, rather they are defined w.r.t. to the object's local frame (or the global frame). AABBRecetpacles could use the AABB center, but not TriangleMeshReceptacles.

I suppose I could add a cached `centroid` variable for any Receptacle and use that, but even then it could be misleading (e.g. if a mesh Receptacle had multiple connected components then the centroid may be off the mesh entirely).